opensrp-family/src/test/java/org/smartregister/family/presenter/BaseFamilyRegisterFragmentPresenterTest.java

```diff
@@ -0,0 +1,104 @@
+package org.smartregister.family.presenter;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.powermock.reflect.Whitebox;
+import org.smartregister.configurableviews.model.Field;
+import org.smartregister.configurableviews.model.RegisterConfiguration;
+import org.smartregister.configurableviews.model.ViewConfiguration;
+import org.smartregister.family.BaseUnitTest;
+import org.smartregister.family.FamilyLibrary;
+import org.smartregister.family.contract.FamilyRegisterFragmentContract;
+import org.smartregister.family.domain.FamilyMetadata;
+
+import java.util.List;
+
+
+/**
+ * @author rkodev
+ */
+public class BaseFamilyRegisterFragmentPresenterTest extends BaseUnitTest {
+
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+
+    private BaseFamilyRegisterFragmentPresenter presenter;
+
+    @Mock
+    private FamilyRegisterFragmentContract.View view;
+
+    @Mock
+    private FamilyRegisterFragmentContract.Model model;
+
+    private String viewConfigurationIdentifier = "12345";
+
+    @Before
+    public void setUp() {
+        presenter = Mockito.mock(BaseFamilyRegisterFragmentPresenter.class, Mockito.withSettings()
```

Comment:
the class under test should not be mocked including during initializing

Reply:
@githengi this mock calls the real methods 
`Mockito.mock(BaseFamilyRegisterFragmentPresenter.class, Mockito.withSettings()
                .useConstructor(view, model, viewConfigurationIdentifier)
                .defaultAnswer(Mockito.CALLS_REAL_METHODS));`

The class `BaseFamilyRegisterFragmentPresenter` is abstract and I needed an instance to test it.